sample_size/sample_size_calculator.py

```diff
@@ -67,27 +45,15 @@ def _get_single_sample_size(self, metric: BaseMetric) -> float:
         return sample_size
 
     def get_sample_size(self) -> float:
-        # Supports the sample size calculation for single metric now.
-        # The current structure is set up to support multiple metrics in the future.
-        sample_size = float("nan")
-        if self.boolean_metrics:
-            for bool_metric in self.boolean_metrics:
-                sample_size = self._get_single_sample_size(bool_metric)
-        if self.numeric_metrics:
-            for numeric_metric in self.numeric_metrics:
-                sample_size = self._get_single_sample_size(numeric_metric)
-        if self.ratio_metrics:
-            for ratio_metric in self.ratio_metrics:
-                sample_size = self._get_single_sample_size(ratio_metric)
+        return self._get_single_sample_size(self.metrics[0])
 
-        return sample_size
-
-    def register_metric(self, metric_type: str, metric_metadata: Dict[str, float]) -> None:
-        VAR_REGISTER_FUNC_MAP = {
-            "boolean": "register_bool_metric",
-            "numeric": "register_numeric_metric",
-            "ratio": "register_ratio_metric",
+    def register_metrics(self, metrics: List[Dict[str, Any]]) -> None:
+        METRIC_REGISTER_MAP = {
+            "boolean": BooleanMetric,
+            "numeric": NumericMetric,
+            "ratio": RatioMetric,
         }
-
-        register_func = getattr(self, VAR_REGISTER_FUNC_MAP[metric_type])
-        register_func(**metric_metadata)
+        for metric in metrics:
+            metric_class = METRIC_REGISTER_MAP[metric["metric_type"]]
+            registered_metric = metric_class(**metric["metric_metadata"])
```

Comment:
you probably miss the suggestion that Dannie commented, would be good. to validate the. metadata:
<img width="1016" alt="image" src="https://user-images.githubusercontent.com/72050506/170388012-3205153e-3f42-47de-bb66-0f85d0e9a21b.png">

Reply:
just did. Here we are validating metrics, a list of dictionaries, instead of the whole experiment config as in the reference, correct?

Reply:
correct. You can move the `METRICS_SCHEMA` outside of `validate_schema`. It is a constant and can be put together with the default parameters at the top of this file.